### ISSUE TITLE
Fix issue with nested interactive elements in the Modal component

### ISSUE THREAD

**Issue by julianguyen** [ifmeorg/ifme#2123]
# Description

In the [Modal component](http://design.if-me.org/?path=/docs/components-modal--text-values-for-element-and-body) ([code](https://github.com/ifmeorg/ifme/blob/main/client/app/components/Modal/index.jsx)), the element that opens the modal has a `role="button"` on it. This causes a problem because the nested element should be an interactive element like a button or link. This will reduce the quality of the screen-reader experience.

We want to remove this role if the [child element](https://github.com/ifmeorg/ifme/blob/main/client/app/components/Modal/index.jsx#L152) has a role and is an interactive element. For safe measure, keep `role="button"` if the child element doesn't have a role and is an interactive element.

---

Please assign yourself (via the Assignees dropdown), if you do want to work on this issue. Can't find yourself? You need to [join our organization](https://github.com/ifmeorg/ifme/wiki/Join-Our-Slack).

Check out our [Picking Up Issues](https://github.com/ifmeorg/ifme/wiki/Picking-Up-Issues) guide if you haven't already!

**Comment by AudreyKj** [comment]
Hello! I just assigned myself and would like to work on this :)

**Comment by julianguyen** [comment]
Awesome, thanks @AudreyKj! Hope you're doing well :)

**Comment by shivansh84ya** [comment]
Hi @julianguyen ,
I would like to work on this issue. Could you please assign it to me?
Thanks!

**Comment by patrikkr858** [comment]
Based on the conversation with @julianguyen  on Slack, I’ve decided to work on this task.
